heartbeat/ibm-cloud-vpc-move-fip.in

```diff
@@ -0,0 +1,248 @@
+    Args:
+        vpc_url (str, mandatory): vpc_url for the region .
+        vni_id_1 (str, mandatory): First VNI(Virtual Network Interface) ID.
+        vni_id_2 (str, mandatory): Second VNI(Virtual Network Interface) ID.
+        api_key (str, optional): IBM Cloud API Access key. Defaults to "".
+
+    Change custom route nexthop to point to this endpoint.
+    In case of a cross AZ Active Passive the route adveritise zone will
+    be chaged to the new acrtive zone
+    """
+    try:
+        active_fip_id , active_fip_ip = ibm_cloud_fail_over.fail_over_floating_ip_start(vpc_url,
+                                                     vni_id_1, vni_id_2, fip_id, api_key)
+    except Exception as e:
+        ocf.logger.error('start_action: Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+    if active_fip_id != fip_id:
+        ocf.logger.error(f'start_action: fip_id: {fip_id} is not attached')
+        return ocf.OCF_ERR_GENERIC
+    ocf.logger.info(f'start_action: OCF_SUCCESS FIP IP: {active_fip_ip} is active')
+    return ocf.OCF_SUCCESS
+
+def monitor_action(
+    vpc_url="",
+    vni_id_1="",
+    vni_id_2="",
+    fip_id="",
+    api_key="",
+):
+    """monitor_action: check if service ip and gateway are responding."""
+    ocf.logger.debug(f'monitor_action: url: {vpc_url} fip_id: {fip_id}' \
+                        f'{vni_id_1}  "vni_id_2:" {vni_id_2}')
+    try:
+        active_fip_id , active_fip_ip = ibm_cloud_fail_over.fail_over_get_attached_fip()
+        if active_fip_id == fip_id:
+            ocf.logger.debug(f'monitor_action: active fip ip: {active_fip_ip}')
+            return ocf.OCF_SUCCESS
+        return ocf.OCF_NOT_RUNNING
+    except Exception as e:
+        ocf.logger.error('Couldn\'t connect with IBM Cloud api: ' + str(e))
+        sys.exit(ocf.OCF_ERR_GENERIC)
+
+
+def main():
+    """Instantiate the resource agent."""
+
+    agent_description = textwrap.dedent("""\
+        Resource Agent to move an IBM Cloud Public Floating IP (FIP) from one virtual network
+        interface (VNI) to another.
+        The prerequisites for the use of this resource agent are as follows:
+
+        1. A two-node (VSI or BM) cluster distributed in same Avilability Zone.
+        2. Enable Instance Metadata on the two nodes.
+        3. Use IBM Cloud API Key or Trused Profile.
+	4. Assign a Floating IP to one VNI.
+    """)
+
+    agent = ocf.Agent(
+        "ibm-cloud-vpc-move-fip",
+        shortdesc="Manages moving an IBM Cloud Public FIP IP",
+        longdesc=agent_description
+    )
+
+
+    agent.add_parameter(
+        "vpc_url",
+        shortdesc="VPC_URL",
+        longdesc="IBM Cloud Public VPC URL for your region or a VPE URL for IBM Cloud VPC",
+        content_type="string",
+        required=True,
+    )
+
+    agent.add_parameter(
+        "vni_id_1",
+        shortdesc="IBM Cloud Virtual Network Interafce UUID for first instance",
```

Comment:
Should be "Interface". Typo also in longdesc and both descs of vni_id_2.

Reply:
Done